### PR TITLE
[php 8.1] Moving from Travis to Github action for php 8.1 support

### PR DESCRIPTION
**Pull request by samsonasik**
@jails this PR try to move from travis CI to github action to add php 8.1 support.

Fixes https://github.com/kahlan/kahlan/issues/389

### CODE REVIEW

**Comment by samsonasik** [comment]
The code `eval'd` by `Double` class seems needs special handling for `#[\ReturnTypeWillChange]`:

```bash
 PHP Fatal error:  During inheritance of ArrayAccess: Uncaught Kahlan\PhpErrorException: `E_DEPRECATED` Return type of Kahlan\Spec\Plugin\Double\Double401::offsetExists(mixed $offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /home/runner/work/kahlan/kahlan/src/Plugin/Double.php(104) : eval()'d code:6
```

@jails could you take a look?

Note:
Code coverage seems fail to upload to codeclimate, seems due to missing config/setting? https://github.com/kahlan/kahlan/runs/4339450106?check_suite_focus=true#step:7:270

File: src/Plugin/Double.php [hunk 1]
```diff
@@ -101,7 +101,13 @@ public static function classname($options = [])
             $code = static::generate($options);
             $nodes = $parser::parse($code);
             $code = $parser::unparse(static::$_pointcut->process($nodes));
+
+            $iniGet = ini_get("error_reporting");
+            ini_set("error_reporting", E_ALL & ~E_DEPRECATED);
+
             eval('?>' . $code);
+
+            ini_set("error_reporting", $iniGet);
```
Comment (samsonasik) [inline src/Plugin/Double.php:110]: @jails this disable `E_DEPRECATED` temporary for eval due generation of code that produce requirement of `#[\ReturnTypeWillChange]`, do you have better solution?

**Comment by samsonasik** [comment]
All checks have passed 🎉 @jails it is ready for review.

**Comment by samsonasik** [comment]
it can be tested in app consumer with:

```bash
composer require --dev kahlan/kahlan:dev-moving-to-github-action
```

File: src/Plugin/Double.php (hunk 1 above)
Comment (samsonasik) [reply src/Plugin/Double.php]: @jails I updated with `preg_replace` so it replace on `(public( static)? function)` generation https://github.com/kahlan/kahlan/pull/392/commits/4fc80b43b516824d8ce2f82191d86cf93e238ab8

**Comment by jails** [comment]
Awesome! Thank you for your effort on this :+1: 
Right now kahlan supports PHP version above 7.1, so I don't think `#[\ReturnTypeWillChange]` is required. I think we just need to set up the correct return type for all functions inherited from PHP core I guess.

**Comment by samsonasik** [comment]
@jails for `eval()`, it seems `#[\ReturnTypeWillChange]` is needed as it generated code, yes ?

**Comment by samsonasik** [comment]
@jails there are methods that need to have `#[\ReturnTypeWillChange]`, eg: `Exception::current()` that extends `Iterator`, it `return mixed` which not exists yet in php 7.1 https://3v4l.org/NETV5#v7.1.33

**Comment by samsonasik** [comment]
@jails I updated to provide return type correctly when supported https://github.com/kahlan/kahlan/pull/392/commits/a2f76c505d1487228fd527452759256203bab9e4 .

Could you handle the upload to `codeclimate` notice? Thank you.

**Comment by samsonasik** [comment]
@jails I updated regex to exclude magic method except `__wakeup` and `__set_state()`, also fix typo for `set_state` intead of `set_sate` https://github.com/kahlan/kahlan/pull/392/commits/41e33f031960aaafafd782cd129a9c9fcf649ff3

**Comment by nwogu** [comment]
Hi @samsonasik @jails. Any updates on this? Anything I can do to help?

**Comment by samsonasik** [comment]
I am waiting @jails for it :)

For note, next to do after it merged:

- make `codeclimate` working, thats possibly need @jails to apply it, or we can move to another service, eg: `codecov`, I wrote a blog post about it https://samsonasik.wordpress.com/2020/02/26/publish-coverage-to-codecov-from-github-actions/

- update README build status badge.

**Comment by phishy** [comment]
🎉

**Comment by samsonasik** [comment]
@jails I need to access `Settings` tab to insert codecov `CODECOV_TOKEN` value, do you want to give me access or you will insert it yourself?

Screen Shot 2022-01-05 at 05 01 22

**Comment by jails** [comment]
@samsonasik I think coveralls & scrutinizer should work. However codeclimate doesn't work anymore, I didn't find the issue. Anyway you're now admin, you can do whatever you want with codecov :smile:

**Comment by samsonasik** [comment]
@jails thank you, I will let you know when everything is done so you can tag a new release.

**Pull request merged**